E2E-Demos/Investment-Banking/Airflow/spark_etl_new.yaml

```diff
@@ -2,63 +2,45 @@ apiVersion: "sparkoperator.hpe.com/v1beta2"
 kind: SparkApplication
 metadata:
   name: "spark-etl"
-  namespace: "spark"
+  namespace: "hpedemo-user01"
 spec:
   sparkConf:
-    # S3 credentials should be specified below
-    spark.hadoop.fs.s3a.endpoint: "https://objectstore-zone1-svc.dataplatform.svc.cluster.local:9000/"
-    spark.hadoop.fs.s3a.access.key: "minioadmin"
-    spark.hadoop.fs.s3a.secret.key: "minioadmin"
-    spark.hadoop.fs.s3a.path.style.access: "true"
+    spark.executorEnv.SPARK_USER: hpedemo-user01
+    spark.hadoop.fs.s3a.access.key: <access-key>
+    spark.hadoop.fs.s3a.secret.key: <secret-key>
+    spark.hadoop.fs.s3a.endpoint: "ua01.hpe-minio-ezaf.com:9000"
     spark.hadoop.fs.s3a.impl: "org.apache.hadoop.fs.s3a.S3AFileSystem"
-    spark.hadoop.fs.s3a.connection.ssl.enabled: "true"
-
-    # Needed if s3 server is behind proxy
-    # spark.hadoop.fs.s3a.proxy.host: "web-proxy.corp.hpecorp.net"
-    # spark.hadoop.fs.s3a.proxy.port: "8080"
-
-    # Needed if s3 server certs are not imported to java truststore used by spark
     spark.driver.extraJavaOptions: "-Dcom.amazonaws.sdk.disableCertChecking=true"
     spark.executor.extraJavaOptions: "-Dcom.amazonaws.sdk.disableCertChecking=true"
+    spark.hadoop.fs.s3a.path.style.access: "true"
+    spark.hadoop.fs.s3a.connection.ssl.enabled: "true"
   type: Python
-  sparkVersion: 3.3.1
+  sparkVersion: 3.4.1
   mode: cluster
-  image: "gcr.io/mapr-252711/spark-py-3.3.1:v3.3.1"
+  image: "gcr.io/mapr-252711/spark-py-3.4.1:v3.4.1"
   imagePullPolicy: Always
-  mainApplicationFile: "local:///mounts/shared-volume/spark/spark_etl_new.py"
+  mainApplicationFile: "local:////mnt/shared/spark/spark_etl_test.py"
   restartPolicy:
     type: Never
   imagePullSecrets:
     - imagepull
   driver:
     labels:
-      version: 3.3.1
-    cores: 1
-    coreLimit: "1"
-    memory: 1g
-    serviceAccount: hpe-spark
+      version: 3.4.1
     volumeMounts:
-      - name: {{dag_run.conf.get("username")}}-volume
-        mountPath: /mounts/{{dag_run.conf.get("username")}}-volume
+      - name: hpedemo-user01-volume
```

Comment:
I believe that Airflow must take care of template resolution, so use templates instead of hardcoding here and below.

Reply:
Remove volumeMounts at all. Airflow is dynamically adding user, shared and datasources volumes for SparkApplication during DAG processing.